scripts/exact-example.jl

```diff
@@ -0,0 +1,237 @@
+
+# We can see that in this case, both lattice-renormalized and transcorrelated Hamiltonians
+# give good results even with modest values of ``M``. Since the values converge as a power
+# law, it's also possible to extrapolate the energies with a fitting package like
+# [LsqFit.jl](https://github.com/JuliaNLSolvers/LsqFit.jl). Keep in mind, however, that for
+# larger systems with very low ``M``, the convergence may not follow a power law.
+
+# ## Speeding up the computations
+
+# As these matrices tend to get large quickly, memory is usually the bottleneck.
+
+# There are currently two methods implemented to reduce the matrix size, `ParitySymmetry`
+# and `TimeReversalSymmetry`. You should only use these where the relevant symmetries
+# actually apply - no checks are performed to make sure they do. To demonstrate them, let's
+# use a Hamiltonian where both do apply. Please consult the documentation for a more
+# in-depth description of these options.
+
+address = FermiFS2C((0,0,0,1,1,0,0), (0,0,1,1,0,0,0))
+ham = HubbardMom1DEP(address)
+
+mat_nosym = Matrix(ham)
+mat_par = Matrix(ParitySymmetry(ham))
+mat_tr = Matrix(TimeReversalSymmetry(ham))
+mat_both = Matrix(ParitySymmetry(TimeReversalSymmetry(ham)))
+
+println("Matrix size")
+println("none:          ", size(mat_nosym))
+println("parity:        ", size(mat_par))
+println("time reversal: ", size(mat_tr))
+println("both:          ", size(mat_both))
+println("\nGroundstate energy:")
+println("none:          ", eigvals(mat_nosym)[1])
+println("parity:        ", eigvals(mat_par)[1])
+println("time reversal: ", eigvals(mat_tr)[1])
+println("both:          ", eigvals(mat_both)[1])
+
+# We see that the ground state for all four options is the same, up to numerical errors,
+# while the dimension of the matrix is reduced by more than half.
+
+# Note that both of these symmetries accept a keyword argument `even` which controls whether
+# even or odd symmetry is applied. If the full spectrum of the Hamiltonian is desired,
+# combine the eigenvalues obtained from using both options.
+
+# If diagonalisation time is a concern, use the `tol` keyword argument to `eigs` or
+# `eigsolve`. Using drastically lower tolerances than default can still produce good results
+# in practice. This, however, should be checked on a case-by-case basis.
+
+# ## Computing observables
+
+# Since all operators in Rimu operate on `DVec`s, we need a way to convert the vectors
+# coming from `eigen` or `eigs` to them. We can use `BasisSetRep.basis` to do that. As an
+# example, let's look at the momentum density.
+
+M = 19
+address = init_address(M)
+bsr = BasisSetRep(HubbardMom1D(
+    address; convert_units(-10, M)..., dispersion=continuum_dispersion
+))
+
+eigvec = eigs(sparse(bsr); nev=1, which=:SR)[2]
+dvec = DVec(zip(bsr.basis, eigvec))
+density_up = [dot(dvec, DensityMatrixDiagonal(i; component=1), dvec) for i in 1:M]
+density_dn = [dot(dvec, DensityMatrixDiagonal(i; component=2), dvec) for i in 1:M]
```

Comment:
I think you need to explain what you use `DensityMatrixDiagonal` for and link to the docstring.